.eslintrc.json

```diff
@@ -3,18 +3,26 @@
     "browser": true,
     "es2021": true
   },
-  "extends": ["eslint:recommended", "plugin:react/recommended"],
+  "extends": [
+    "eslint:recommended",
+    "plugin:react/recommended",
+    "plugin:@typescript-eslint/eslint-recommended",
+    "plugin:@typescript-eslint/recommended"
+  ],
+  "parser": "@typescript-eslint/parser",
   "parserOptions": {
     "ecmaFeatures": {
       "jsx": true
     },
     "ecmaVersion": 2021,
     "sourceType": "module"
   },
-  "plugins": ["react"],
+  "plugins": ["react", "@typescript-eslint"],
   "rules": {
-    "no-unused-vars": "warn",
+    "no-unused-vars": "off",
+    "@typescript-eslint/no-unused-vars": ["warn"],
```

Comment:
within type definitions that define a function with parameters, the usual eslint no-unused-vars would complain. I've turned that off in favour of the typescript no-unused-vars rule. See https://stackoverflow.com/questions/67411154/eslint-no-unused-vars-on-function-parameters-within-typescript-types-interfa for more info